### PR TITLE
Marked Antiforgery cookie as essential

### PR DESCRIPTION
**Pull request by kichalla**
Related to https://github.com/aspnet/Home/issues/2408 : Determine if the antiforgery cookie is considered essential

### CODE REVIEW

File: src/Microsoft.AspNetCore.Antiforgery/AntiforgeryOptions.cs [hunk 1]
```diff
@@ -20,6 +20,7 @@ public class AntiforgeryOptions
         {
             SameSite = SameSiteMode.Strict,
             HttpOnly = true,
+            IsEssential = true,
```
Comment (Tratcher) [inline src/Microsoft.AspNetCore.Antiforgery/AntiforgeryOptions.cs]: Comment that it's essential because otherwise form posts won't work?

**Review by Tratcher — APPROVED** [review]

File: src/Microsoft.AspNetCore.Antiforgery/AntiforgeryOptions.cs [hunk 2]
```diff
@@ -45,6 +46,8 @@ public class AntiforgeryOptions
         /// <para>
         /// <see cref="CookieBuilder.SameSite"/> defaults to <see cref="SameSiteMode.Strict"/>.
         /// <see cref="CookieBuilder.HttpOnly"/> defaults to <c>true</c>.
+        /// <see cref="CookieBuilder.IsEssential"/> defaults to <c>true</c>.
```
Comment (rynowak) [inline src/Microsoft.AspNetCore.Antiforgery/AntiforgeryOptions.cs]: This block might be a good place to put an explanation.

**Review by kichalla — COMMENTED** [review]
🆙 📅

File: src/Microsoft.AspNetCore.Antiforgery/AntiforgeryOptions.cs [hunk 3]
```diff
@@ -45,6 +48,10 @@ public class AntiforgeryOptions
         /// <para>
         /// <see cref="CookieBuilder.SameSite"/> defaults to <see cref="SameSiteMode.Strict"/>.
         /// <see cref="CookieBuilder.HttpOnly"/> defaults to <c>true</c>.
+        /// <see cref="CookieBuilder.IsEssential"/> defaults to <c>true</c>. If this is marked as <c>false</c>, then
+        /// this cookie token is not sent in the response and any further requests to the server which require
+        /// antiforgery validation would fail due to this missing cookie token.
```
Comment (rynowak) [inline src/Microsoft.AspNetCore.Antiforgery/AntiforgeryOptions.cs]: This isn't really accurate.

I'd suggest something like:
``
The cookie used but the antiforgery system is part of a security system that is necessary when using cookie-based authentication. It should be consider required for the application to function.
```

**Pull request merged**